serde_derive/src/internals/ast.rs

```diff
@@ -153,8 +154,12 @@ fn enum_from_ast<'a>(
                 fields,
                 original: variant,
             }
-        })
-        .collect()
+        }).inspect(|variant| {
+            if !variant.attrs.untagged() && seen_untagged {
+                cx.error_spanned_by(&variant.ident, "all variants with the #[serde(untagged)] attribute must be placed at the end of the enum")
```

Comment:
Why only allowing `#[serde(untagged)]` to be at the end?

Reply:
This keeps the implementation more consistent with the deserialisation strategy of going though each variant in order and choosing the first variant that matches.